tests/crypto_test.py

```diff
@@ -274,8 +307,12 @@ def test_signed_only_false(self):
 
     @staticmethod
     def _context_mock():
-        error = gpgme.GpgmeError()
-        error.code = gpgme.ERR_AMBIGUOUS_NAME
+        # We need an uninittialized class that is a descendent of GPGMEError
+        # for the test, but we need to make the getcode method a mock, so we
+        # use type to make a dynamic sublcass, since raise wont' accept
+        # anything that isn't an Exception
+        error = type('CustomError', (gpg.errors.GPGMEError, ), {})
+        error.getcode = mock.Mock(return_value=gpg.errors.AMBIGUOUS_NAME)
```

Comment:
I did not know the three arguments form of `type()` and was expecting some deep magic because I was suspecting that there is some kind of thing that is an "unitialized class" that can not be created with the `class` keywoed but only with `type()`.

I vote for a "normal" class definition bacause it comunicates what is happening more obviously. (Maybe I am the only one who didn't know `type()` all that well).